### PR TITLE
Add ability to get the time and percentage elapsed/remaining in a billing cycle

### PR DESCRIPTION
**Pull request by brianpattison**

### CODE REVIEW

**Comment by brianpattison** [comment]
[Billing Cycle gem improvements (1d)](https://trello.com/c/z0l7Avfu/401-billing-cycle-gem-improvements-1d)

File: README.md [hunk 1]
```diff
@@ -59,6 +66,58 @@ billing_cycle.previous_due_at
...
+```ruby
+original_billing_date = Time.zone.parse("2019-06-01 00:00:00")
+billing_interval = 1.month
+billing_cycle = BillingCycle::BillingCycle.new(original_billing_date, billing_interval)
+
+Time.zone.now
+# => Sun, 16 Jun 2019 00:00:00 CDT -05:00
+
+billing_cycle.time_elapsed
+# => 1296000.0
```
Comment (bmedenwald) [inline README.md]: Add a note that this is seconds.

File: README.md [hunk 2]
```diff
@@ -59,6 +66,58 @@ billing_cycle.previous_due_at
...
+
+billing_cycle.time_elapsed(1.day, Time.zone.parse("2019-06-07 00:00:00"))
```
Comment (bmedenwald) [inline README.md]: Add a note that 1.day is asking how many days have elapsed... and from the 7th of the month.

File: README.md [hunk 3]
```diff
@@ -59,6 +66,58 @@ billing_cycle.previous_due_at
...
+# => 6.0
+
+billing_cycle.time_remaining
+# => 2073600.0
```
Comment (bmedenwald) [inline README.md]: Are these the wrong number?

File: README.md [hunk 4]
```diff
@@ -59,6 +66,58 @@ billing_cycle.previous_due_at
...
+# => 0.5
+
+billing_cycle.percent_elapsed(Time.zone.parse("2019-06-07 00:00:00"))
+# => 0.2
+
+billing_cycle.percent_remaining
+# => 0.5
+
+billing_cycle.percent_remaining(Time.zone.parse("2019-06-07 00:00:00"))
+# => 0.2
```
Comment (bmedenwald) [inline README.md:128]: Wrong number?

File: README.md [hunk 5]
```diff
@@ -59,6 +66,58 @@ billing_cycle.previous_due_at
...
+
+```ruby
+original_billing_date = Time.zone.parse("2019-06-01 00:00:00")
+billing_interval = 1.month
+billing_cycle = BillingCycle::BillingCycle.new(original_billing_date, billing_interval)
+
+Time.zone.now
+# => Sun, 16 Jun 2019 00:00:00 CDT -05:00
+
+billing_cycle.percent_elapsed
```
Comment (bmedenwald) [inline README.md:117]: The percentage as a fraction of 1.0

**Review by bmedenwald — CHANGES_REQUESTED** [review]

**Review by bmedenwald — APPROVED** [review]

**Pull request merged**